### PR TITLE
Use Yargs for commandline arguments

### PR DESCRIPTION
**Pull request by SimenB**
Instead of handling CLI parsing ourselves, I think delegating to `yargs` is better.

The problem right now I think is that commandline-parsing is coupled to the whole app, instead of just passing along options for the main module to handle.

So instead of fighting with the tests only parsing options passed the first time (as `yargs` caches `process.argv`) and the process quitting if printing out version or help, I moved this logic into the `bin/stylint`-file, and init takes `options` as a parameter.

Version and help is only from the CLI, the main app doesn't care about that, thus I removed the tests for it. Add new tests for the CLI on just that? Unsure if it's necessary.

Output for `help` is now this, btw. Much nicer, imo!

image

There is currently one test failing but that is failing on develop as well.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -20,8 +20,8 @@
   "main": "index.js",
   "files": [
     "bin/",
-    "src/",
-    "index.js"
+    "index.js",
```
Comment (SimenB) [inline package.json]: Changes other than added `yargs` and `lodash.defaults` is npm sorting keys and strings

File: src/core/flags.js [hunk 2]
```diff
@@ -1,11 +0,0 @@
-// flags for the cli
```
Comment (SimenB) [inline src/core/flags.js]: Was this flag used?

**Comment by rossPatton** [comment]
Awesome, definitely will need to review this. :+1:

File: src/utils/setConfig.js [hunk 3]
```diff
@@ -11,33 +11,25 @@ var stripJsonComments = require( 'strip-json-comments' )
  * 2: user passes location of .stylintrc file to use
  * 3: user has a .stylintrc file in their dir but doesnt pass anything
  * 4: none of the above, fallback to initial config
+ * @param {String} [configpath] If defined, the path to a config-file to read
  * @returns {Function} kick off linter again
 */
-var setConfig = function() {
+var setConfig = function( configpath ) {
```
Comment (SimenB) [inline src/utils/setConfig.js]: Might be more correct to set it on `this.state` or something like that? As I've mentioned before, I've not touched stampit before, unsure what conventions there are :smile:

**Pull request merged**